jetstream/core/metrics/prometheus.py

```diff
@@ -35,27 +35,146 @@ def __new__(cls):
       documentation="Size of prefill queue",
       labelnames=["id"],
   )
+
   _transfer_backlog = Gauge(
       name="jetstream_transfer_backlog_size",
       documentation="Size of transfer queue",
       labelnames=["id", "idx"],
   )
+
   _generate_backlog = Gauge(
       name="jetstream_generate_backlog_size",
       documentation="Size of generate queue",
       labelnames=["id", "idx"],
   )
+
+  _queue_duration = Histogram(
+      name="jetstream_queue_duration",
+      documentation="The total time each request spends enqueued in seconds",
+      labelnames=["id"],
+      buckets=[
+          0.01,
+          0.02,
+          0.05,
+          0.1,
+          0.2,
+          0.5,
+          1.0,
+          2.0,
+          5.0,
+          10.0,
+          20.0,
+          50.0,
+          100.0,
+      ],
+  )
+
   _slots_used_percentage = Gauge(
       name="jetstream_slots_used_percentage",
       documentation="The percentage of decode slots currently being used",
       labelnames=["id", "idx"],
   )
+
   _server_startup_latency = Gauge(
       name="jetstream_server_startup_latency",
       documentation="Total time taken to start the Jetstream server",
       labelnames=["id"],
   )
 
+  _time_to_first_token = Histogram(
+      name="jetstream_time_to_first_token",
+      documentation="Time to first token per request in seconds",
+      labelnames=["id"],
+      buckets=[
```

Comment:
Do you have a reference for the interval setting? Why do we set different values of interval for the bucket?

Reply:
[To agree with how vllm sets their buckets for TTFT](https://docs.vllm.ai/en/latest/serving/metrics.html#:~:text=self.histogram_time_to_first_token,%2C%2010.0%0A%20%20%20%20%20%20%20%20%20%20%20%20%5D))